### PR TITLE
NullableTypeTransformer - constructor property promotion support

### LINKED ISSUE #5401: `TernaryOperatorSpacesFixer` incompatible with nullable constructor property promotion
## Bug report

After upgrading to PHP8 I have problems with CS Fixer. `TernaryOperatorSpacesFixer` inserts space after `?` that marks nullable arguments in constructors with property promotion:

```diff
--- Original
+++ New
@@ -7,7 +7,7 @@
 class Foo
 {
     public function __construct(
-        private ?string $foo = null,
+        private ? string $foo = null,
     ) {
         is_null(true ? '...' : null);
     }
```

### Code snippet that reproduces the problem

```php
<?php

class Foo
{
    public function __construct(
        private ?string $foo = null,
    ) {
        is_null(true ? '...' : null);
    }
}
```

**NOTE**: Ternary operator used anywhere in the class triggers this invalid change. Without actual ternary in the code, constructor is not touched. Without property promotion it works properly too.

### PR DESCRIPTION
**Pull request by Wirone**
Fixes #5401

### CODE REVIEW

File: tests/Tokenizer/Transformer/NullableTypeTransformerTest.php [hunk 1]
```diff
@@ -135,4 +135,39 @@ public function provideProcess74Cases()
...
+    }
+
+    public function provideProcess80Cases()
+    {
+        return [
+            [
+                '<?php
+                    class Foo
+                    {
+                        public function __construct(private ?string $foo = null)
```
Comment (keradus) [inline tests/Tokenizer/Transformer/NullableTypeTransformerTest.php:163]: as we have explicit list of private/protected/public in source file, can we add 3 variants here, please?

Comment (Wirone) [reply tests/Tokenizer/Transformer/NullableTypeTransformerTest.php]: @keradus Good point :+1: Fixed

**Review by julienfalque — APPROVED** [review]

**Comment by SpacePossum** [comment]
@Wirone thanks for reporting and fixing, looking good!

Could you add your example from https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5401 as a test to [TernaryOperatorSpacesFixerTest](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.17.3/tests/Fixer/Operator/TernaryOperatorSpacesFixerTest.php) (with `@requires PHP 8.0`) ?

**Review by keradus — APPROVED** [review]
really nice contribution!

**Comment by keradus** [comment]
Thank you @Wirone.

**Comment by Wirone** [comment]
@keradus happy to hear that! And thank you for the great tool :+1:

**Pull request merged**